src/MongoDB.Driver/MongoCredential.cs

```diff
@@ -524,6 +536,30 @@ private static MongoCredential FromComponents(string mechanism, string source, s
                         mechanism,
                         new MongoInternalIdentity(source, username),
                         evidence);
+                case "MONGODB-AWS":
+                    // MUST be "$external". Defaults to $external.
+                    EnsureNullOrExternalSource(mechanism, source);
+                    if (username == null)
+                    {
+                        if (evidence != null && evidence is PasswordEvidence)
+                        {
+                            throw new ArgumentException("A MONGODB-AWS must have access key id.");
```

Comment:
I'm not sure about this exception. I will rely on @rstam  opinion here

Reply:
Ditto above: should this "A MONGODB-AWS credential must have an access key ID."? Perhaps we should use a constant?